### PR TITLE
more metadata dev - introduce keymap

### PR DESCRIPTION
**Pull request by jihyunbak**
NOTE: This work builds on top of two open PRs #41 and #45. The temporary base branch `tmp_base` is the expected state of the main branch after these two prior PRs are merged. I will rebase onto main once those PRs are merged.

The major change made by this PR is the use of resource files
- `metadata_keymap.yaml` and 
- `metadata_keymap_legacy.yaml`.

Keymap for the new metadata reflects recent discussion with @jthermiz , and the input relies on the newly parsed metadata yaml file. (With two manual edits: commented out the `.nan` entry between experiment and block metadata, and added a block metadata entry `stimulus`.)

Hopefully this could make the metadata structure more readable and easily editable...

### CODE REVIEW

**Comment by jihyunbak** [comment]
@jthermiz It might be helpful to actually check out this branch, go to the `tests/` folder, and run the metadata tests by doing `python -u test_metadata_manager.py`. Intermediate yaml files at three checkpoints (input, input-clean, full) will be created under a temporary directory `_test/`. It would be great if you could inspect these yaml files, to make sure things make sense to you!

**Comment by jthermiz** [comment]
> @jthermiz It might be helpful to actually check out this branch, go to the `tests/` folder, and run the metadata tests by doing `python -u test_metadata_manager.py`. Intermediate yaml files at three checkpoints (input, input-clean, full) will be created under a temporary directory `_test/`. It would be great if you could inspect these yaml files, to make sure things make sense to you! 

Hi Ji Hyun, I went through the metadata_full output. It looks good but I do have the following comments:

Is the weight unit pre-defined in nwb? The units of weight are in grams

I know this is legacy stuff. I don't think it's useful information so maybe get rid off it's not being used for anything?
  stim_values: np.ones(60)
  baseline_start: 0.3
  baseline_end: 0.5
  first_mark: 0.0
  luigi_cls: WhiteNoiseAnalysis
  tokenizer: wn_tokenize

Based on Ryan's feedback, I added additional details. Since the ecog grid is so large relative to to brain regions, perhaps this will suffice. Generally speaking the ecog grid encompasses A1 and other auditory areas. Based on these details the end user should be able to infer this.

location_details: right temporal lobe --> the most posterior lateral electrode is 1 mm from lateral ridge and 3 mm from posterior ridge 


For this block, poly was false so should we not include the poly device details?
In the future if one uses poly but not ecog should we not include ecog device details?

**Comment by jthermiz** [comment]
Wrt to the legacy data, I think it makes sense to put the ecog and poly electrode location in the same coordinate system. Currently, they are in separate  coordinates based on the R56 yaml

File: nsds_lab_to_nwb/metadata/exp_note_reader.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
+import pandas as pd
```
Comment (jthermiz) [inline nsds_lab_to_nwb/metadata/exp_note_reader.py:1]: @jihyunbak Added exp_note_reader.py and changed yaml to write file using block_folder

Comment (jihyunbak) [reply nsds_lab_to_nwb/metadata/exp_note_reader.py:1]: Thanks @jthermiz !

**Comment by jihyunbak** [comment]
Merging this for now; further improvement to be made in separate PRs.

**Comment by jihyunbak** [comment]
@jthermiz let's keep the discussion going - I made a bunch of issues referencing your comments here. Thanks!

**Pull request merged**